### ISSUE TITLE
ctl.tokenIsExpired(...) segfaults in (remote) netauth clients

### ISSUE THREAD

**Issue by Johnnynator** [netauth/netauth#12]
Currently reproducible for me on all my systems with these steps:

Adding config in `~/.netauth/config.toml` and cert to `~/.netauth/keys/tls.pem`, get token with `netauth auth get-token`.
The token will be saved as expected to `/tmp/john.nt`.

After this netauth segfaults (when running on a remote device) when running `inspect-token`, `validate-token` and probably others.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x92e89d]

goroutine 1 [running]:
github.com/netauth/netauth/internal/ctl.tokenIsExpired(...)
        /builddir/netauth-0.4.2/internal/ctl/helpers.go:56
github.com/netauth/netauth/internal/ctl.token(0x0, 0xc000187cd0)
        /builddir/netauth-0.4.2/internal/ctl/helpers.go:43 +0xbd
github.com/netauth/netauth/internal/ctl.authInspectTokenRun(0xe2b360, 0xe662e8, 0x0, 0x0)
        /builddir/netauth-0.4.2/internal/ctl/auth-inspecttoken.go:38 +0x34
github.com/spf13/cobra.(*Command).execute(0xe2b360, 0xe662e8, 0x0, 0x0, 0xe2b360, 0xe662e8)
        /builddir/netauth-0.4.2/_build-NetAuth-xbps/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xe2aea0, 0xe66750, 0xe380e0, 0x0)
        /builddir/netauth-0.4.2/_build-NetAuth-xbps/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fe
github.com/spf13/cobra.(*Command).Execute(...)
        /builddir/netauth-0.4.2/_build-NetAuth-xbps/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/netauth/netauth/internal/ctl.Execute()
        /builddir/netauth-0.4.2/internal/ctl/root.go:79 +0x31
main.main()
        /builddir/netauth-0.4.2/cmd/netauth/main.go:39 +0x25f
```

**Comment by the-maldridge** [comment]
I am unable to reproduce this locally, can you confirm that in addition to the TLS public key you also have an RSA public key for the token verification in your keys directory?

**Comment by Johnnynator** [comment]
Yes, the rsa pub key is there
```
ls ~/.netauth/keys/
tls.pem  token.pem
```

**Comment by Johnnynator** [comment]
Okay, I now figured out what causes this, there is a check to make sure that the permission mask is exactly `644`, the file was `640` on my system. It would probably better to just check if it is at least as restrictive and not requiring that it is exact `rw-r-r`

https://github.com/netauth/netauth/blob/722d1965a88e5725f79d275ceb067ce1bb3f0d6a/internal/token/jwt/rsa.go#L150-L153

**Comment by the-maldridge** [comment]
https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/os/types.go;l=23 seems to suggest that it is not possible to portably check for permission bits.  Additionally, the returned error should always manifest in a logged error message, so further investigation is necessary to determine why you received a segfault instead of an error.  If I had to take an immediate guess, I'd guess that the CLI is not handling things correctly with the returned error.

The entire token code is up for refactoring in the next major release, so I'd like to leave this issue open until that has shipped.

**Comment by the-maldridge** [comment]
I have taken an alternate approach, and have simply removed the permissions check since there isn't really a good way to do it portably, and it doesn't add significant security value.  This will be resolved in the next major release.

**Comment by the-maldridge** [comment]
Resolved in release v0.6.0
